### ISSUE TITLE
Accepted party invite, but Invitation doesn't disappear

### ISSUE THREAD

**Issue by AquelePedro** [HabitRPG/habitica#5649]
I was invited to Brasil Team party and accepted, but the invitation remains

Already left party and accepted invitation again, but it didn't make the invitation go away.
Tried that using Chrome and Firefox @ Windows and then Chrome and Firefox @ Linux, still the same result.

_edit by admin:_

**The information below will no longer work. If you can't remove a party invitation, please go to Help > Report a Bug from the Habitica website's menu, or Guilds > Report a Bug on the mobile app (both take you to the same guild). An admin will delete the invitation when they see your message.**

**Alternatively you can post on this issue if you wish. Even though it's closed, we'll still see your message and help you. If you post here, you do not need to also report it in the guild.**

Follow the steps below exactly as they are listed.

If at any time during these steps, you see the red error message box on the right hand side of the page, reload the page, and try to continue the instructions from where you were up to (or from the start if that makes things easier).
1. Go to the party page.
2. Decline the invitation.
3. IMMEDIATELY clear your browser's cookies for habitica.com. Don't do ANYTHING else between declining the invitation and clearing the cookies (don't log out, don't reload, nor anything else). 
4. Reload Habitica. You'll find you're logged out (if you aren't, then clear cookies again).
5. Log in. 
6. If the login doesn't work (e.g., the login form doesn't disappear after you click the login button) then reload the page and log in again. This second login should work.
7. If the invitation is still there, go to your party page and decline it again.
8. IMMEDIATELY clear your browser's cookies for habitica.com again. Don't do ANYTHING else between declining the invitation and clearing the cookies.
9. Reload Habitica, log in again.
10. You should now find that the invitation is gone permanently. If it is not, double-check that you followed exactly those instructions, and if you are certain that you did, comment here to tell us that it didn't work. If anything happened that didn't seem to match what these instructions say, please describe it.

If you still have an email that contains an invitation link for a party, delete the email. Do not click on that link again.

Those instructions should permanently delete the invitation that keeps asking you if you want to join the party. However your name will still be listed in the list of people invited to the party (as well as in the list of party members) but we are considering that to be a lower priority bug since it is less annoying. We are sorry about the inconvenience.
##

**Comment by Alys** [comment]
We'll need your user ID to help with this. If you don't know how to get your User ID, go to Help -> Report a Bug and see step 6.

**Comment by AquelePedro** [comment]
Here it is:

4e3e2f5f-42b8-4aef-9059-8c8c07cd2a1b

**Comment by SpenserJ** [comment]
Does this go hand-in-hand with active members still appearing in the invite list?
![Glitchy member list](https://cloud.githubusercontent.com/assets/90011/8811091/a60c4190-2fb0-11e5-9a7c-76499c5bb18c.png)
Sammy7131 (highlighting is mine) appears in both the invited and member list of my party, and I've had a few other users appear this way. When I removed an inactive member (also in the invite list), he disappeared from the member list, but was still listed as invited. After removing him the second time, all trace of the user was finally gone from the lists. Sammy7131's ID is 9b044b09-909e-415f-8c67-495246489cf4.

**Comment by Alys** [comment]
@Tsvet I've removed the invitation from your account. Please tell us if that hasn't fixed the problem. Do you know if the invitation was sent by specifying your email address or your user ID?

@SpenserJ Yes, that probably is related. Did you send the invitation to Sammy7131? Do you recall if it was by email or ID? In Sammy7131's account, there is still an active invitation so I would be interested to know if that person appears back in the invite list.

**Comment by SpenserJ** [comment]
@Alys I invited Sammy7131 via ID, and it was the same for the other users who appeared in both lists (sadly I didn't write down their names). I'll keep an eye out for this happening again, and try to reproduce it locally. I'm not seeing anything in the code ([website/src/controllers/groups.js - Line 434-483](https://github.com/HabitRPG/habitrpg/blob/7223eeb6815c056a76532726318037688cf8b791/website/src/controllers/groups.js#L434-L483)) that stands out as potentially flawed.

**Comment by Alys** [comment]
Another report of this was sent by email (below). I've deleted the invitation.

Hi, I was messing around and somehow got a notification (invitation to a guild) that won’t go away. Would you delete it for me? It puts a little person icon in the messages status, so I don’t see when I have new chat messages. L

My UUID:  9173198f-4777-4a35-9f03-8aea24f51782

"invitations": {
    "party": {
      "id": "2fdfb274-3286-439f-b8ec-c480f7066405",
      "inviter": "9173198f-4777-4a35-9f03-8aea24f51782",
      "name": "Students of the Bear"
    },
    "guilds": []
  },

Strangely, this is a guild; not a party. Not sure why it’s under party and not guild…

How this happened: I was trying to figure out if the email invitation link could just be sent to anyone, or if it was email address specific. In the process, I guess I invited myself to my own guild a couple of times. I hit the decline invitation button, and that got rid of one notification, but not the other. I tried leaving that guild, but the notification remains.

Thanks!
Edward Shapard (Sir Edward)

**Comment by Taejang** [comment]
We had two players appearing as both active and invited. As the party leader, I tried to remove a party invitation, but it instead removed the player from both the invite section and the active list, booting him from the party entirely. After another invite, that player shows as active and not under invited. The second user still shows as both invited and active.

The first time invites were sent to both players, it was by email. The second time (after the player was booted from the party), I added him back with the ID. A third player was invited via email and, after accepting, was successfully added to the active list and removed from the invited list.

**Comment by religigeek** [comment]
I have been experiencing this issue as well. I joined via an email link from a friend two days ago. Signed in, accepted the invitation, then the invitation still appears. Tried multiple browsers on multiple computers and it still persists.

User ID: 7993e524-9734-4d3d-8471-e9bab192d716
Party: I Like Cake

**Comment by Alys** [comment]
@religigeek I can't see any invitations saved in your account or in that party. Please tell us if the problem still appears at your end.

**Comment by religigeek** [comment]
It looks like it finally disappeared. I will update if it shows up again.

**Comment by swistak** [comment]
Same issue here  user id: a847788d-5f96-4bf3-94c5-7e1ac7b4cb0c
Joined via email link to.

This is still happening and not resolved for me. group id: 8a9fb119-fce7-40a9-93d7-bac37a4f0d42

See also ticket #6051

**Comment by Alys** [comment]
@swistak There's no invitations remaining in your user account or in the party you're a member of. Please log out of Habitica, clear your browser's cache, and local storage, close all browser windows, and then see if the invitation has disappeared. Sorry about the trouble! I think it's just a local caching/storage issue.

**Comment by swistak** [comment]
Yea. Seems this has been solved with other ticket and it remained in cache for some reason.

Looks like there's a problem with angular loading twice when you enter habitica through deep link (eg. #/tasks ). I'm trying to track it down, to create reproducible example.

All this trouble just reinforces my opinion that angular is just horrible, undebugable pile of garbage.

**Comment by dirad** [comment]
i just had this issue.
Invited a new user which logged in via facebook, and had this hanging invite issue.
They also showed up twice in my list. Once as Member and once as invited.

They could only get rid of it after i deleted them from the invites..

**Comment by cTheDragons** [comment]
Hi,

We are also experience the same issue with our party Group ID: e3c001ca-f0dd-4510-b313-ba38dbfc7294

for the user 
asympt0te d36de64c-b5d5-4aba-a3be-6df41c39efa6

They are appearing both in the party list and the invited list.

(We are also concern that the others on the invited list have already reject the invite but they are still 'HANGING around')

Also have logged into multiple computers and browser and still see the invite.

Thanks
cTheDragons

**Comment by Alys** [comment]
@cTheDragons Can you please ask the party leader to cancel the invitations? They might have to do that a couple of times before they disappear permanently. Please tell us if that doesn't work.

**Comment by cTheDragons** [comment]
Hi Alys,

Thanks @Alys for your quick response. 
I am the party leader (sorry for not being clear).

With  the invite for user asympt0te **_d36de64c-b5d5-4aba-a3be-6df41c39efa6**_ they are already part of the party and the quest. I'm concern if I cancel the invite I will kick them from the group, particular in light of the comments above.

With the other invitees I will cancel myself. (I will wait 24 hours to _ensure_ they actually don't want to be apart of the group).

Thanks again.
cTheDragons

**Comment by Alys** [comment]
@cTheDragons I've deleted the invitation for d36de64c-b5d5-4aba-a3be-6df41c39efa6.

**Comment by cTheDragons** [comment]
Thanks @Alys  All seems to be good.

**Comment by chamini2** [comment]
Hello, I'm not the party leader of my party, but I see there's three invites for three people already in the members list, the Group ID is fbd39690-cb18-4f65-9a92-55a781e133a3

Hope someone can give me a hand fixing this?

**Comment by Alys** [comment]
@chamini2 I've removed the invitations. It's not unheard of for them to come back, so comment here again if they do.

**Comment by JamesOwers** [comment]
I have the same issue. Was invited to a party by multiple members and now the invitation persists even though I am already a member. I'm also still on the 'invited' list. uuid=36b11762-799c-4719-8b66-a10c6da05c5e

**Comment by Alys** [comment]
@kungfujam I've fixed that for you, but it's not unheard of for it to come back. If it does, comment here again and also clear local storage for your browser (you might need to google for how that's done, for some browsers it's different than clearing the cache).

**Comment by JamesOwers** [comment]
Cheers, that's worked but only after again accepting the invitation.

**Comment by danielvancamp** [comment]
Same issue here, both joined and were invited by mail. 
Group ID :  793119dc-6589-4dff-8282-1ffd3dd32671

**Comment by Alys** [comment]
@danielvancamp Affected users are 5d91a8fb-4781-4f1d-a040-a10f58c99348 and 0ce5bf69-77cc-4c14-9ab5-97a5ed128ac6. I've removed the invitations for them, but they often come back a few times. I'll check again a couple of times a day for the next few days and remove them again if needed. If you notice the same problem for those users a week from now, please comment again here.

**Comment by swistak** [comment]
Maybe you should use those cases to locate a root cause for this?

**Comment by Alys** [comment]
@swistak yes, that's what we've been doing. It's most likely to be a local storage issue, and we'll be investigating removing or adjusting the use of local storage when a large project to significantly improve the code base is finished.

**Comment by pseudocyesis** [comment]
Hello! I'm having the same problem with the invite not disappearing. ID below! Thanks :)

d7f1e549-d4ef-43b5-a277-934a9f7b9a21

**Comment by Alys** [comment]
@pseudocyesis  I've removed the invitation, but it's not unknown for it to come back a few times. I'll check again a couple of times a day for the next few days and remove it again if needed. If you notice the same problem a week from now, please comment again here.

**Comment by erica3219** [comment]
http://i.imgur.com/YrOLL5y.jpg
having same problem here as well.
UUID: 9bab6b19-9cea-4f1c-b9e8-5c54c61a549d
Thanks

**Comment by Alys** [comment]
@erica3219 I've removed the invitation, but it's not unknown for it to come back a few times. I'll check again a couple of times a day for the next few days and remove it again if needed. If you notice the same problem a week from now, please comment again here.

**Comment by erica3219** [comment]
As well, I attend a party, but I cast none damage to the monster. I mean, I am cleared my to dos and dailies, however, there was neither text coming up said how much damage I cast nor the blood on monster really gone. But I can use skills as usual.

**Comment by Alys** [comment]
@erica3219 That won't have been caused by the invitation. There's nothing wrong in your account for the party's quest, so I don't think there's a bug. If the quest has only just started, no one will have damaged it yet. Damage is dealt by each player only once a day (when they first use Habitica each day). If you need extra information about this, please go to Help -> Ask a Question from the Habitica website, and we'll give you more details there.

**Comment by erica3219** [comment]
Thanks a lot :+1:

**Comment by pseudocyesis** [comment]
Hey @Alys, still seeing the invitation on my account, hoping to see if you had any new tricks up your sleeve! Thanks

ID again: d7f1e549-d4ef-43b5-a277-934a9f7b9a21

**Comment by Alys** [comment]
@pseudocyesis Yes, it keeps appearing. :( I've seen it and removed it several times over the past week. I'm checking twice every day and I'll keep doing that until I see it stop coming back for a few days in a row. I'm afraid it's just a matter of waiting until it clears properly. When we finish some major site upgrades, we'll look into whether our use of local storage can be improved, which I think is what this is related to. I'm really sorry about the trouble!

**Comment by jpazniokas** [comment]
I just accidentally booted a player mid-quest by attempting to delete a non-disappearing invite, as per this bug.

Out of curiosity, is there any way to get a player back into a quest after being invited back into the party?  It's unfortunate: the player in question has done the bulk of the damage over the last two days.

**Comment by Alys** [comment]
@jpazniokas There isn't normally but if you give us their uuid after they are back in the party, we can put them back in the quest, since this happened as a result of a bug. We would not usually do that in non-bug circumstances.

**Comment by jpazniokas** [comment]
Thank you so much, @Alys .  

b291b934-7d0b-4936-8818-4daeab575211

I appreciate it.  If it were a simple matter of XP, gold and items, I would have just hacked the rewards together myself.  But there are also two pieces of equipment and an achievement involved too.  :)

**Comment by Alys** [comment]
@jpazniokas They should be back in the quest now. Please reload your own party page and look at the list of quest participants. Tell me if they don't appear or if anything looks odd. Check tomorrow after they have been through cron to see if a message appears in party chat about their quest progress. Tell me then if that's not right either.

You have another player in the party with this invitations problem. I've just deleted their excess invitation. It might come back a few times, so I'll be checking twice a day for the next few days until it's deleted for good (FYI, I have this scripted so it's very little trouble to do). If there are still invitations left in a week, comment here again.

**Comment by jpazniokas** [comment]
Thank you, @Alys , and not a moment too soon: little did I know that another party member dumped all of their mana into Brutal Smashes: the quest boss had a mere two hours left to live! 

I'll let you know about superfluous invites if you like, but I'm a programmer too: this is a pretty low-priority one with everything else y'all have going on, and it's pretty easy to work around: just ignore it.

Thanks again.

**Comment by JFDama** [comment]
Hi @Alys, a party I joined recently, group UUID 1ea2408e-4c4e-4aeb-aa29-924a350ad0fc, is seeing this problem for most of our joined members as well. Would you mind taking care of this for us?

The UUIDs of the players with this issue in the party are
e7f03b14-5a7f-4539-98b7-0eec0b92872a
ec1989c9-51ab-4099-9225-7ae8d0a21f8c
7a9669fc-ddb8-41d8-b294-57b51164ba32
9349b978-c25e-40a8-8e8d-2d3ed6e0d5e6
e1e87003-5771-44a1-b5fe-60796e64df8f
a99b33dc-b030-4bed-8573-d16b6847cff3
5a1bf377-517d-4c2c-b2a3-13b3f89294fa

Thanks!

**Comment by Alys** [comment]
@JFDama I've deleted those invitations. They typically come back several times over a few days so I'll keep checking at least twice a day and delete them when I see them. It's unusual to get so many at once, so I'd like to work out why. Do you know if all those people were invited at the same time? And were they all invited by the same person? Can you tell me who that was?

**Comment by JFDama** [comment]
@Alys, thank you! 

These people were not all invited at the same time or all by the same person. In fact, it might be the opposite--I joined later than most, so I can't report the full history, but 9349b978-c25e-40a8-8e8d-2d3ed6e0d5e6 invited me, e1e87003-5771-44a1-b5fe-60796e64df8f, two weeks ago, then I invited a99b33dc-b030-4bed-8573-d16b6847cff3 3 days after, and he invited 5a1bf377-517d-4c2c-b2a3-13b3f89294fa yesterday. So the part of the history I know has been more like a sequential chain than a broadcast, and weeks have gone between some of these invitations.

**Comment by Alys** [comment]
@JFDama Well that blows my theory out of the water. :) Thank you very much for the details!

**Comment by JFDama** [comment]
@Alys Thank you again for the fix!

**Comment by jozefspisiak** [comment]
Same bug here. I accepted invitation and it sticks, even if I accept it multiple times afterwards.
User ID: 700a11ad-ee57-461e-b6f8-8af962e79f1c
Group ID: 79bfa2eb-9778-4187-97c9-6d6aeb45b523

Other members of group are as well affected.

**Comment by Alys** [comment]
@jozefspisiak I've deleted the invitations, but please read the bold text at the top of this page, if you haven't already.

**Comment by jozefspisiak** [comment]
@Alys yes, thank you. The invitation still shows up, but as I read they will reappear for couple of days. Can you maybe check access logs, if they are enabled to see if there were multiple ajax calls which created them or only one? If the problem is in the browser or only on the server side. 

Also we have all registered to habitica after we received email invitation to join party from a friend, which came: Wed, 6 Jan 2016 09:36:49 -0800 (PST) and my user was created Thu, 7 Jan 2016 00:44:41 -0800 (PST) and I first time accepted invitation few minutes afterwards. Just though this might be important.

**Comment by ghost** [comment]
Hello!

I have a lingering invitation for someone in my party.
User ID: e93cfb63-b562-486c-9bff-4aaddb8bed1d
Group ID: a9a60bd3-0922-4c2f-82b9-08144dac995b
Any assistance would be greatly appreciated.

Also, if it's helpful, the person with the duplicate invitation was invited by email and joined the party first. The more recent member had an account already, and does not appear to have any issues.

Thanks!

**Comment by Alys** [comment]
@thomasohashi I've deleted the invitations, but please read the bold text at the top of this page, if you haven't already (in brief; the invitation is likely to reappear a few times but it will stop after a few days). 

@jozefspisiak and @thomasohashi : Thanks for the extra information, although unfortunately the joining method doesn't seem to be consistent - we've had reports of it happening from established users who received invitations by User ID too.

@jozefspisiak It has to be happening in the browser; I can't think of anything on the server that could be causing this.

**Comment by shiboe** [comment]
Howdy, have a member that also shows up in the invited section. 
My user ID: 35da2eae-af20-4bb7-95b5-a96cf5cc0e60
Duplicated user ID: b586f0bd-fce8-44ea-8ab8-ddf4f3ef6596
Group ID: aeda929f-044c-4f99-b050-f13be9aa076f

**Comment by Alys** [comment]
@shiboe I've deleted the invitation, but please read the bold text at the top of this page, if you haven't already (in brief; the invitation is likely to reappear a few times but it will stop after a few days).

**Comment by pjcunnin** [comment]
I have a group invitation that won't go away. Any help would be greatly appreciated.
User ID: 0b6c9ca3-9d18-4cfc-96d8-b1af784a22cc
Group ID: c048bc89-c9c3-4871-ab18-e3680549f6a3

**Comment by Alys** [comment]
@pjcunnin I've deleted the invitation for you and the other two party members with the same problem, but please read the bold text at the top of this page, if you haven't already (in brief; the invitations are likely to reappear a few times but will stop after a few days).

**Comment by Vcarrus** [comment]
Hi! I'm having this same issue, but as an invite for my own party? I've read the bold text!

User ID: 4fabc9f2-3c9d-42f3-917a-6280f38abaae
Group ID: e08c6ad0-ac94-4152-a764-ee3f9bf17a84

**Comment by Alys** [comment]
@Vcarrus  :)  Done!

**Comment by Bootporeon** [comment]
Hello, I just joined today, and I've been invited to a party by a good friend; we are also having this problem.

My User ID: 5817d4c4-98f3-40ee-9019-41fc12b271c5

The group ID: cbd08c44-86f0-4002-b0b2-e25f44ea0a4d

I also have a few other errors I'm getting when I refresh the party page, but I don't know if they're related; I am not sure if the fact that I cannot attack the boss my party is fighting is one.

**Comment by Alys** [comment]
@Bootporeon I've deleted the invitation, but please read the bold text at the top of this page, if you haven't already (in brief; the invitation is likely to reappear a few times but it will stop after a few days).

The other errors are probably not related to this (but probably would be related to the other bug report you logged, which I've just replied to).

The reason you're not attacking the party's quest boss is because you're not a participant on the quest. Either the quest started before you joined the party, or the person who sent the quest invitations started the quest before you had a chance to accept the invitation. That person can, if they wish, choose to abort the quest. The quest scroll will go back into their quest page for them to use again. If you or they have questions, see the wiki's Quests page at http://habitica.wikia.com/wiki/Quests or drop in to Help > Ask a Question from the Habitica website's menu (there's a lot of friendly people there who'll be happy to answer any specific questions you might have).

**Comment by SimplyNaOH** [comment]
Having this issue.

User ID of the member that's not being removed from invited: d416d605-71e7-4dcf-bdda-a4095196b509

My User ID: 1a29d6ef-bd86-400a-8276-0decdbf1d429

Group ID: 1ce204f1-c213-40ae-8e84-5e1df88e9af6

Hope you can solve it, thanks!

**Comment by Alys** [comment]
@SimplyNaOH That's fixed but please see the bold text at the top of this issue (in brief, the invitation will keep coming back for a few days before it disappears).

**Comment by nbunney** [comment]
User ID: 767b1e2a-3b74-496a-aa53-732525b12f9c
Group ID: 7ab54905-8b85-4d4f-a299-c7a69407510e

Appreciate the help on this!

**Comment by Alys** [comment]
@nbunney that's fixed for now, but please see the bold text at the top of this issue (in brief, the invitation will keep coming back for a few days before it disappears).

**Comment by nbunney** [comment]
Actually it does not appear to be fixed, unless it comes back immediately.  

So I went back to the site and then to the party page and clicked the sync button to see if it would remove the invite from the page.  I got an application error when I synced.  Refreshing the page removed the error but the invite is still there.
<img width="1427" alt="screen shot 2016-02-10 at 3 08 32 pm" src="https://cloud.githubusercontent.com/assets/1700382/12965067/6a18d9a2-d008-11e5-8d98-8c85cf9130ca.png">

**Comment by Alys** [comment]
@nbunney Yes, they can come back immediately - sorry about that! I've removed it again now. If you notice it again, decline it and then immediately log out of Habitica - I'd be interested to see if that prevents it coming back ever again. In the meantime, I'll keep clearing it at least twice a day.

**Comment by nbunney** [comment]
Thanks @Alys, I saw it immediately and so I declined and logged out immediately as suggested and then logged back in.  It was back immediately.  I have never not seen that message on the party screen.

**Comment by nbunney** [comment]
@Alys,  Not sure if this is related at all but I am getting errors on refresh right now.

Request URL:https://static.xx.fbcdn.net/rsrc.php/v2iaFb3/yE/l/en_US/TILZkhqC-Bs.js
Request Method:GET
Status Code:400 Service Worker Fallback Required (from ServiceWorker)

Request URL:https://habitica.com/api/v2/user/batch-update?_v=114&data=1455205501410
Request Method:POST
Status Code:503 Service Unavailable
Remote Address:45.79.106.117:443

Two of these:
Request URL:https://habitica.com/api/v2/groups/party
Request Method:GET
Status Code:503 Service Unavailable
Remote Address:45.79.172.232:443

**Comment by david--wright** [comment]
My party and I are experiencing this issue:
Party ID
46287d8e-b67c-45f6-b290-ec90b85a21e3
User ID's
My user ID:
ea8e3a93-91a2-42ec-9a84-c70e08942c6a
Other User ID's with issue
3b427eb3-bb9e-4376-8e53-9095c9d50cc9
e0a1d4f0-d450-48b2-970d-ffbe16b41dc1
e12c8ad0-6af7-474b-a3ec-e359dbb71d45

Even though I created the party I show in the invited section as do the other 3 even though they have accepted the invitations. I invited two of the other users the third was invited by one of the users I invited. All three of the invites were sent within 10 minutes or so of setting up the party. All of these invites were done via user ID. An email invite was also sent to another person. They waited a couple of days to join and joined via the android mobile app. Logging in to the website they did not have an invitation listed. I re-invited them via user ID - They joined the party and are no longer listed in the invited section of the party screen. This was 10 minutes ago so I don't know if the issue will affect them. The other four still show in the invited section this user does not so I am guessing it won't.
Their user ID is:
bb40eab3-210e-44bf-aa7c-29574bee46c4

If there is anything I can do to help troubleshoot the issue let me know.

**Comment by Alys** [comment]
@david--wright that's fixed for now, but please see the bold text at the top of this issue (in brief, the invitation will keep coming back for a few days before it disappears).

@david--wright and @nbunney Thanks for the details! I'll read over them again tomorrow and see if I can find some clues.

**Comment by DiscipleGeek** [comment]
I'm also having the same issue, as well as tons of 503 errors upon refreshing any page on the site.

UserID: ee2bc6de-6fba-4835-a6c8-8cd3c00d8809

**Comment by Alys** [comment]
@DiscipleGeek   If it's still showing the errors, try logging out of Habitica. Clearing cookies might also help.

**Comment by PixelPuppy** [comment]
Same issue as well as tons of errors on pages to do with party members. (takes a long time to load, and then errors out. No error number though)

UserID: 4ddf5412-6797-4438-9aa7-6398f9be7ae2
GroupID: 2cc11023-3ce0-462e-a571-d1696b101052

**Comment by Alys** [comment]
@PixelPuppy that's fixed for now, but please see the bold text at the top of this issue (in brief, the invitation will keep coming back for a few days before it disappears).

**Comment by david--wright** [comment]
@Alys Just a quick update - I logged out of habatica on all affected machines and then cleared local storage and didn't ever see the invitation come back. On one of the other accounts in our party we cleared local storage without logging out and the issue did come back once - we then logged out cleared local storage and logged back in and didn't see the issue after that.

**Comment by quance** [comment]
@Alys I'm in the same situation.  Invited to join a party.  Accepted.  received a second invitation that I declined.  But the second invitation keeps popping up over and over again.  Tried to decline and logout immediately.  Still there.  Party page then routinely gives an update error when reloaded.

UserID:  7fb581d3-a528-4380-adbf-37ce5055d7f9

thanks

**Comment by Alys** [comment]
@quance I've removed the invitation but see the text in bold at the top of the page (in brief; the invitation is likely to reappear a few times but it will stop after a few days).

**Comment by quance** [comment]
Cheers

**Comment by Alys** [comment]
For admins:

Here's a couple of things I've noticed (I've added them to the top post since they'll get buried here):
- The inviter (i.e., the person who supposedly sent the invitation) is sometimes the same as the person who received it, even though it shouldn't be possible to invite yourself to a party.
- The invitation will sometimes list an old name of the party, not the party's current name, strongly implying that the invitation is not being sent by the site but instead is cached in the user's browser from an earlier time.

**Comment by Kainzaah** [comment]
The same problem. 
Party ID: 7bced89c-ff5d-4718-9e83-ffca3eff8397
User ID: 598b1c44-4abe-45cc-ad04-82e3ce910e7b

Other users also have the problem.

Thank you :)

**Comment by PixelPuppy** [comment]
@Alys hello. 7 days have passed, and the problem is still persisting. :(
I tried logging out and clearing cookies/cache/history. Didn't work..

UserID: 4ddf5412-6797-4438-9aa7-6398f9be7ae2
GroupID: 2cc11023-3ce0-462e-a571-d1696b101052

**Comment by Alys** [comment]
@Kainzaah  I've removed the invitations but see the text in bold at the top of the page (in brief; the invitation is likely to reappear a few times but it might stop after a few days).

@PixelPuppy Darn. :( Thanks for telling me. I'll write some code in Habitica to delete the invitations every time they would otherwise appear to you. I was hoping to avoid that but the current process is clearly not working well enough.

**Comment by joaopkato** [comment]
Not only I have this problem, but I was removed from the party I was in for no reason(It wasn't anyone in the party, it was a system bug). Also because I was removed from the party, I wasn't able to complete the quest we were doing(Ochtothulu's Call), so I lost all the rewards we were gonna get upon completion.

If possible I'd like this problem fixed and I'd like to be put back in the quest, because I feel is unfair that I lose out on this quest because of a system problem.

My ID: 3b3caf1d-9408-4fc8-ac91-b9e299624be1
Party ID: 728f09d3-5f47-481c-8739-58c107a37ced

**Comment by quance** [comment]
Alys

This is still occurring and causing loading errors on the party page. It's
been several days with no change. Not sure if there is a better fix

**Comment by Alys** [comment]
@quance Sorry about that. If it's really getting on your nerves, using a different browser might be less annoying.  I think we're going to have to change the code so that it doesn't show invitations that aren't appropriate.

@herolaubet How long ago did the quest start? Are you back in the party now? If so, how did you get back into it - did you accept an invitation, or did you reload the party page and found that you were back in the party with no need to accept an invitation?

**Comment by Alys** [comment]
For everyone who is still suffering from this bug, try the steps below. I'm sorry it's such a pain, but it's important to follow these instructions exactly.

If at any time during these steps, you see the red error message box on the right hand side of the page, reload the page, and try to continue the instructions from where you were up to (or from the start if that makes things easier).
1. Go to the party page.
2. Decline the invitation.
3. IMMEDIATELY clear your browser's cookies for habitica.com. Don't do ANYTHING else between declining the invitation and clearing the cookies (don't log out, don't reload, nor anything else). 
4. Reload Habitica. You'll find you're logged out (if you aren't, then clear cookies again).
5. Log in. 
6. If the login doesn't work (e.g., the login form doesn't disappear after you click the login button) then reload the page and log in again. This second login should work.
7. If the invitation is still there, go to your party page and decline it again.
8. IMMEDIATELY clear your browser's cookies for habitica.com again. Don't do ANYTHING else between declining the invitation and clearing the cookies.
9. Reload Habitica, log in again.
10. You should now find that the invitation is gone permanently. If it is not, double-check that you followed exactly those instructions, and if you are certain that you did, comment here to tell us that it didn't work. If anything happened that didn't seem to match what these instructions say, please describe it.

If you still have an email that contains an invitation link for a party, delete the email. Do not click on that link again.

**Comment by Alys** [comment]
I forgot to add:

Those instructions should permanently delete the invitation that keeps asking you if you want to join the party. However your name will still be listed in the list of people invited to the party (as well as in the list of party members). That's less annoying though. We'll do something about that separately.

We'll also adjust the code to make these invitations disappear by themselves, but I'm not sure when we'll be able to finish that.

**Comment by joaopkato** [comment]
@Alys I received an email saying the quest had started 18 hours ago(according to gmail). The email was sent Feb, 25 at 3:50 PM at my time zone (UTC -03:00 - Brasília).
At 5:16 PM I received another email saying I was removed from the party. When I logged in via Firefox I was out of the party but had an invitation still, so I accepted and got back in. It was trough invitation.

**Comment by Alys** [comment]
@herolaubet I'd like to see the email saying you were removed from the party. Can you please forward it to admin@habitica.com?

**Comment by joaopkato** [comment]
Done

**Comment by Alys** [comment]
@herolaubet That email is typically sent only when a party leader has removed a player from the party. I'm not aware of any bugs that cause it to be sent at other times - which is not to say there aren't any, it's just that since we don't know of any bugs, we need to be cautious about this. If you'd still like to be put back into the quest, please ask your party leader to send me a Private Message in Habitica to state that they approve of you being added to the quest. The reason for the PM in Habitica is so that I can confirm from their profile that they are the party leader.

**Comment by Alys** [comment]
@herolaubet You're back in the quest now. Reload the party page if you don't see that change yet.

**Comment by Alys** [comment]
I believe this can no longer happen since Habitica's API v3 rewrite.
